diff --git a/node_modules/gatsby-plugin-no-javascript/gatsby-ssr.js b/node_modules/gatsby-plugin-no-javascript/gatsby-ssr.js
index 5484ebc..da96ab2 100644
--- a/node_modules/gatsby-plugin-no-javascript/gatsby-ssr.js
+++ b/node_modules/gatsby-plugin-no-javascript/gatsby-ssr.js
@@ -44,6 +44,14 @@ function getHeadComponentsNoJS(headComponents, pluginOptions) {
         if (headComponent.props.href && (headComponent.props.href.startsWith('/static/') || headComponent.props.href.startsWith('/page-data/'))) {
             return false;
         }
+        // This is the CSS added automatically by gatsby-remark-autolink-headers
+        if (headComponent.key === 'gatsby-remark-autolink-headers-style') {
+            return false;
+        }
+        // This is the Javascript added automatically for a better SPA scroll experience: gatsby-remark-autolink-headers
+        if (headComponent.key === 'gatsby-remark-autolink-headers-script') {
+            return false;
+        }
         return pageScripts.find(function (script) {
             return headComponent.props.as === 'script' &&
                 "/" + script.name === headComponent.props.href &&
@@ -64,6 +72,9 @@ function getPostBodyComponentsNoJS(postBodyComponents, pluginOptions) {
         if (postBodyComponent.props.id && (postBodyComponent.props.id === 'gatsby-script-loader' || postBodyComponent.props.id === 'gatsby-chunk-mapping')) {
             return false;
         }
+        if (postBodyComponent.props &&  postBodyComponent.props.src && postBodyComponent.props.src.startsWith('/polyfill')) {
+            return false;
+        }
         return pageScripts.find(function (script) { return postBodyComponent.type === 'script' && "/" + script.name === postBodyComponent.props.src; }) === undefined;
     });
 }
