diff --git a/node_modules/gatsby-plugin-scss-typescript/gatsby-node.js b/node_modules/gatsby-plugin-scss-typescript/gatsby-node.js
index 373c285..80ebd62 100644
--- a/node_modules/gatsby-plugin-scss-typescript/gatsby-node.js
+++ b/node_modules/gatsby-plugin-scss-typescript/gatsby-node.js
@@ -10,17 +10,14 @@ exports.onCreateWebpackConfig = ({ stage, plugins, rules, actions }, options) =>
         camelCase: true,
         namedExport: true,
         ...cssLoaderOptions,
-    })
-        .use.map(loader => {
-        if (!isProduction) {
-            ['/css-loader/locals', '/css-loader/', '\\css-loader\\locals', '\\css-loader\\'].forEach(modulePath => {
-                if (loader.loader.includes(modulePath)) {
-                    loader.loader = require.resolve('typings-for-css-modules-loader');
-                }
-            });
-        }
-        return loader;
-    });
+    }).use;
+
+    // We we use a different typings loader that supports the newest css-loader
+    // And we always install it, not just in production mode
+    const miniCSS = useCssModules.shift();
+    useCssModules.unshift(require.resolve('@teamsupercell/typings-for-css-modules-loader'));
+    useCssModules.unshift(miniCSS);
+
     const sassLoaders = [
         { loader: 'resolve-url-loader' },
         {
